src/connector/src/source/base.rs

```diff
@@ -560,6 +564,8 @@ pub trait SplitMetaData: Sized {
     fn encode_to_json(&self) -> JsonbVal;
     fn restore_from_json(value: JsonbVal) -> Result<Self>;
     fn update_with_offset(&mut self, start_offset: String) -> crate::error::ConnectorResult<()>;
+
+    fn get_encoded_offset(&self) -> String;
```

Comment:
Would you mind adding some docs to the interfaces in this trait? For example, what's the difference (behavior and usage) between `encode_to_json` and `encoded_offset`?

Reply:
done